_stbt/core.py

```diff
@@ -681,11 +681,20 @@ def new_device_under_test_from_config(
     if transformation_pipeline is None:
         gst_sink_pipeline = get_config('global', 'transformation_pipeline')
 
-    display = Display(
-        gst_source_pipeline, gst_sink_pipeline, _mainloop(), save_video,
-        restart_source, transformation_pipeline)
+    display = [None]
+
+    def raise_in_user_thread(exception):
+        display[0].tell_user_thread(exception)
```

Comment:
You don't need to make `display` be an array because the only place we access it from a different context (here) is for calling one of its methods, not for *setting* `display[0]`.